app/src/main/kotlin/io/homeassistant/companion/android/widgets/todo/TodoWidgetConfigureActivity.kt

```diff
@@ -145,11 +145,14 @@ class TodoWidgetConfigureActivity : BaseActivity() {
     }
 
     private fun onAddWidget() {
-        viewModel.prepareData()
-        // TODO show the error when it makes sense
-        setResult(RESULT_OK)
-        viewModel.updateWidget(this)
-        finish()
+        try {
+            viewModel.addWidgetConfiguration()
+            setResult(RESULT_OK)
+            viewModel.updateWidget(this)
+            finish()
+        } catch (_: IllegalStateException) {
+            showAddWidgetError()
+        }
```

Comment:
I don't really like using exception to manage the error. If you have any other suggestion I'm all hears.